lib/puppet/util/windows/registry.rb

```diff
@@ -312,6 +312,13 @@ def reg_delete_key_ex(key, name, regsam = KEY64)
       result
     end
 
+    def sanitize(value)
+      # Replace byte sequences that do not compose
+      # to valid UTF8, and null bytes with the Unicode
+      # Replacement Character: "�"
+      return value.scrub!.gsub(/\0/, "\uFFFD")
+    end
```

Comment:
So to follow up on the comments from Josh, I think this can actually just be simplified down to:

```ruby
def sanitize(value)
  # Replace null bytes with a space
  value.gsub!("\x00", ' ')
end
```

Note use of `gsub!` and use of a literal null rather than a regex.

This gives us what I think we ultimately want:

```
[6] pry(main)> "foo\x00\x00\u0000".gsub!("\x00", ' ')
=> "foo   "
```

Reply:
is better to use `gsub` because `gsub!` will result in `nil` if there is nothing to replace

Reply:
In @Iristyle's example, you'd probably want more something like:
```ruby
def sanitize(value)
  # Replace null bytes with a space
  value.gsub!("\x00", ' ')
  value
end
```
I think that they were suggesting using `gsub!` instead of `gsub` to avoid creating a new string (since `gsub!` will modify the existing one).